### PR TITLE
Add  PS1 Script for EntityFramework Migrations

### PR DESCRIPTION
**Pull request by Lordapa**

### CODE REVIEW

File: ApplyMigrations.ps1 [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+function ApplyMigration(){
+   Write-Output "Start Applying Migrations"
+   $lastMigration = Get-ChildItem -Path $dir -Exclude *ModelSnapshot.cs | Sort-Object LastAccessTime -Descending | Select-Object -First 1 
+   Write-Output $lastMigration.name
+   dotnet ef database update --startup-project  $startUp --context $contextDb --verbose 
+   Write-Output "Finishing Applying Migrations"
+}
+
+if((docker inspect donateto_postgres).count -eq 1)
```
Comment (nicolasgranata) [inline ApplyMigrations.ps1:14]: if-else statement should be 

```
if () {
} 
else {
}
```

File: ApplyMigrations.ps1 [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+{
+   Write-Output "Container does not Exist."
+   Write-Output "Start PostgreDb Build."
+   docker-compose up -d --build postgres_donateto
+   Write-Output "Finishing PostgreDb Build."
+}else
```
Comment (nicolasgranata) [inline ApplyMigrations.ps1:20]: maybe we can remove this else and just use the if statement written below

File: ApplyMigrations.ps1 [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+{
+    Write-Output "Checking if PostgreDb is UP."
+    if((docker inspect -f '{{.State.Running}}' donateto_postgres) -eq 'false' )
+    {
+       Write-Output "Starting Container."
+       docker-compose up -d postgres_donateto
+    }
+}
```
Comment (nicolasgranata) [inline ApplyMigrations.ps1:28]: We should put the if logic into a function and just call it, maybe in the line 38

File: ApplyMigrations.ps1 [hunk 4]
```diff
@@ -0,0 +1,39 @@
+$dir = 'src\DonateTo.Infrastructure\Data\Migrations'
+$migrations = Get-ChildItem -Path $dir -Recurse -Filter *.cs
+$contextDb = "DonateToDbContext"
+$startUp = '.\src\DonateTo.WebApi\DonateTo.WebApi.csproj'
+
+function ApplyMigration(){
+   Write-Output "Start Applying Migrations"
+   $lastMigration = Get-ChildItem -Path $dir -Exclude *ModelSnapshot.cs | Sort-Object LastAccessTime -Descending | Select-Object -First 1 
+   Write-Output $lastMigration.name
+   dotnet ef database update --startup-project  $startUp --context $contextDb --verbose
```
Comment (nicolasgranata) [inline ApplyMigrations.ps1:10]: What happens if we have more than one migration to apply?

File: ApplyMigrations.ps1 [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+       docker-compose up -d postgres_donateto
+    }
+}
+
+Write-Output "Container UP."
+Write-Output "Checking for Migrations"
+
+if (($migrations | Measure-Object).Count -eq 0)
+{
+    Write-Output "No Migrations founded"
```
Comment (guillemurano) [inline ApplyMigrations.ps1:35]: Replace by "No migrations found."

File: ApplyMigrations.ps1 (hunk 2 above)
Comment (Lordapa) [reply ApplyMigrations.ps1:20]: if we remove it, its gonna make and unnecesary inspect of the container

Comment (nicolasgranata) [reply ApplyMigrations.ps1:20]: maybe the logic should be changed for something like, if container exists and it's not running --> start it, else create it

File: ApplyMigrations.ps1 [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+   dotnet ef database update --startup-project  $startUp --context $contextDb --verbose 
+   Write-Output "Finishing Applying Migrations"
+}
+
+CheckContainerStatus
+if (($migrations | Measure-Object).Count -eq 0) {
+   Write-Output "No migrations found"
+}
+else {
+   ApplyMigration
```
Comment (nicolasgranata) [inline ApplyMigrations.ps1:32]: function check should be called here in order to not execute it if there is no migrations

**Pull request merged**